### PR TITLE
Small refactor in test service for ec2 to include the possibility of SEV testing

### PR DESCRIPTION
**Pull request by apozsuse**
Could not finish this code change. I'll retake the job after the workshop...

### CODE REVIEW

File: .virtualenv.requirements.txt [hunk 1]
```diff
@@ -19,7 +19,7 @@ APScheduler
 python-dateutil>=2.6.0
 amqpstorm
 ec2imgutils>=9.0.3
-img-proof>=7.14.0
+img-proof>=7.21.0
```
Comment (apozsuse) [inline .virtualenv.requirements.txt:22]: Included this change to assure `img-proof` has the new param.

File: mash/services/test/ec2_job.py [hunk 2]
```diff
@@ -42,16 +42,24 @@
         'm5.large',
         't3.small'
     ],
+    'hybrid-sev': [
+    ],
```
Comment (apozsuse) [inline mash/services/test/ec2_job.py:46]: Pending.
We'd need to check which instance_types we want to test for SEV_CAPABLE images when UEFI is supported.

**Comment by apozsuse** [comment]
This one is ready for review.

File: mash/services/test/ec2_job.py [hunk 3]
```diff
@@ -127,31 +122,49 @@ def run_job(self):
...
+
+            if info['partition'] in ('aws-cn', 'aws-us-gov') and \
+                    self.cloud_architecture == 'aarch64':
+                # Skip test aarch64 images in China and GovCloud.
+                # There are no aarch64 based instance types available.
+                continue
+
+            if all([
+                info['partition'] == 'aws',
+                'SEV_CAPABLE' in self.guest_os_features,
```
Comment (rjschwei) [inline mash/services/test/ec2_job.py:135]: 'SEV_CAPABLE' is a Google setting, it is a bit confusing to see it pop up in the EC2 code stream.  Is the plan to set this in the job docs for EC2? And how is this hen documented in a sensible way for other users (not us) of mash?

Comment (apozsuse) [reply mash/services/test/ec2_job.py:135]: Yes, the plan is to set those settings in the job docs where suitable.
I reused the Google setting names but I'm ok with changing these settings to match AWS terminology to make them more explicit:
  - I'll change 'guest_os_features' in the job doc for 'cpu_options'
  - 'SEV_CAPABLE' for 'AMD_SEV_SNP_ENABLED' 
  
WDYT?

Comment (smarlowucf) [reply mash/services/test/ec2_job.py:135]: Yeah, I agree on using aws specific terminology here.

File: mash/services/test/ec2_job.py [hunk 4]
```diff
@@ -52,6 +57,10 @@
     ]
 }
 
+sev_capable_regions = [
+    'us-east-1'
```
Comment (smarlowucf) [inline mash/services/test/ec2_job.py:61]: The feature is only available in us-east-2 and eu-west-1 regions.

**Review by smarlowucf — CHANGES_REQUESTED** [review]

**Review by smarlowucf — APPROVED** [review]

File: mash/services/test/ec2_job.py [hunk 5]
```diff
@@ -127,31 +123,49 @@ def run_job(self):
...
+                # There are no aarch64 based instance types available.
+                continue
+
+            if all([
+                info['partition'] == 'aws',
+                'AMD_SEV_SNP_ENABLED' in self.cpu_options,
+                region in sev_snp_supported_regions,
+                self.cloud_architecture == 'x86_64'
+            ]):
+                self.sev_snp_enabled = True
```
Comment (rjschwei) [inline mash/services/test/ec2_job.py:140]: Come to think of it I do not think this should be a flag, i.e. true/false. We should expect TDX and we will have to test TDX instances as well. What if we change the name of the variable to `mem_encryption` with a value of  `SEV` , `TDX`, or  `NONE` and then the `select_instance_type` method can switch on that setting and either pick from `hybrid-sev-snp` (I think that could maybe simply be `sev-snp-instances`, not sure why `hybrid` is supposed to express) or the future `tdx-instances`. Then the condition can then be expressed as:
```
self.mem_encryption = 'NONE'
if info['partition'] == 'aws' and self.cloud_architecture == 'x86_64':
    if 'AMD_SEV_SNP_ENABLED' in self.cpu_options and region in sev_snp_supported_regions:
        self.mem_encryption = 'SEV'
    if 'INTEL_TDX_ENABLED' in self.cpu_options and region in tdx_supported_regions:
        self.mem_encryption = 'TDX'
```
And of course this really applies to all CSPs as such this should really be lifted to the base class of the jobs. A bit more abstraction is needed for that. It's a bit bigger change but we need the same logic for all 3 CSPs. The requirement can be boiled down to selecting a specific instance type in a specific region when either TDX or SEV is set and run that instance with a specific flag. This is common across all 3 CSPs we support.

**Review by rjschwei — COMMENTED** [review]
I am not opposed to merging this and coming back with a follow up PR. That said, we are not in a hurry, SEV-SNP in AWS is currently broken and we need to get that fixed before we enable this testing.

**Comment by apozsuse** [comment]
Closing this PR as this will be implemented in the test refactor.

**Pull request closed without merging**